src/Servers/Kestrel/test/InMemory.FunctionalTests/LoggingConnectionMiddlewareTests.cs

```diff
@@ -14,6 +14,7 @@ namespace Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests
     public class LoggingConnectionMiddlewareTests : LoggedTest
     {
         [Fact]
+        [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/34561")]
```

Comment:
> | [6.375s] Microsoft.AspNetCore.Server.Kestrel.Connections Debug: Connection id "0HMAB2I8425HF" accepted.
> | [10.059s] Microsoft.AspNetCore.Server.Kestrel.Connections Debug: Connection id "0HMAB2I8425HF" started.

> [20.479s] Microsoft.AspNetCore.Server.Kestrel.Https.Internal.HttpsConnectionMiddleware Debug: Authentication of the HTTPS connection timed out.

It looks like there might have been some serious threadpool starvation for the "accepted" and "started" logs to be ~4 seconds apart. The only thing between those two logs is a threadpool dispatch. I think that's what caused the handshake timeout and failed test.

Accepted log:

https://github.com/dotnet/aspnetcore/blob/67deada4fb1caea2fcf7ae275e7350a6bb4b3a75/src/Servers/Kestrel/Core/src/Internal/ConnectionDispatcher.cs#L61-L64

Which immediately dispatches to the startup log:

https://github.com/dotnet/aspnetcore/blob/67deada4fb1caea2fcf7ae275e7350a6bb4b3a75/src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelConnectionOfT.cs#L32-L46

I'm not sure we should be quarantining tests because of threadpool starvation caused by who-knows-what. This seems to just be the unlucky victim.